### DISCUSSION TITLE
Data necessary for proving?

### DISCUSSION THREAD

**Discussion by JoseSK999** [ZeroSync/ZeroSync#55] [Q&A] [answered]
Hey guys, congrats for this project. 

I'm wondering what data is necessary to create a proof of valid sync, taking into account recursive proofs. Let's say I want to prove the state at block 300k, but there's already a proof at block 250k. Do I need only the blocks between 250k and 300k (plus state at 250k and its proof)?

Or do I also need a proof for every block in between?

**Comment by lucidLuckylee** [comment, marked as answer]
Hi,
I don't get if you are referring to the proof generator or the client perspective in your question. So here is my for answer both:

_Client_ - you run a bitcoin full node and want to sync it to block 300k:
The proof for block 300k includes all previous proofs (so also the proof for block 250k) due to the recursive verification. You wouldn't need any other proof than the one for the block you want to sync at. Whether you start from block 250k or any other block is irrelevant. 

_Proof generator_ - you want to provide a proof for block 300k:
Assuming you already proved every block from genesis to 250k and in each step recursively aggregated them into a single proof, you can start at 250k and have to prove _**every**_ block until you end up at block 300k.

**Reply by JoseSK999** [reply]
So the prover, in this case, doesn't need the blocks up to the 250kth right? She only needs to have knowledge of the blocks that she will prove (250k-300k).

**Reply by lucidLuckylee** [reply]
Yes
